core/runtime/impl/wasm_memory_stream.cpp

```diff
@@ -0,0 +1,35 @@
+#include <utility>
+
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "runtime/impl/wasm_memory_stream.hpp"
+
+namespace kagome::runtime {
+
+  WasmMemoryStream::WasmMemoryStream(std::shared_ptr<WasmMemory> memory)
+      : memory_(std::move(memory)), current_ptr_{0} {}
+
+  bool WasmMemoryStream::hasMore(uint64_t n) const {
+    return current_ptr_ + n <= memory_->size();
+  }
+
+  std::optional<uint8_t> WasmMemoryStream::nextByte() {
+    if (not hasMore(1)) {
```

Comment:
If I remember correctly, somebody told me that we agreed to use !, && and || instead of keywords, when I tried to use them, so I will just left it here.

Reply:
In offline discussion we decided that we can any approach